lib/Doctrine/ORM/EntityManager.php

```diff
@@ -157,8 +156,12 @@ protected function __construct(Connection $conn, Configuration $config, EventMan
         $this->metadataFactory->setCacheDriver($this->config->getMetadataCacheImpl());
 
         $this->repositoryFactory = $config->getRepositoryFactory();
-        $this->unitOfWork        = new UnitOfWork($this);
-        $this->proxyFactory      = new ProxyFactory(
+
+        $unitOfWorkClassName  = $config->getUnitOfWorkClassName();
+        $this->unitOfWork     = new $unitOfWorkClassName($this);
```

Comment:
@mcspronko it should be in a separate PR. And you should explain clearly *why* you need this. The UnitOfwork is quite sensible logic, so I'm not sure it makes sense to make it configurable.